gtsfm/loader/olsson_loader.py

```diff
@@ -0,0 +1,154 @@
+"""Simple loader class that reads Carl Olsson's dataset from a folder on disk.
+
+Authors: John Lambert
+"""
+
+import glob
+import os
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+from gtsam import Cal3Bundler, Pose3, Rot3
+from scipy.io import loadmat
+
+import gtsfm.utils.io as io_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class OlssonLoader(LoaderBase):
+    """Simple loader class that reads Carl Olsson's dataset from a folder on disk.
+
+    Ref: http://www.maths.lth.se/matematiklth/personal/calle/dataset/dataset.html
+
+    Folder layout structure:
+    - RGB Images: images/
+    - Intrinsics + Extrinsics data (optional): data.mat
+
+    If explicit intrinsics are not provided, the exif data will be used.
+    """
+
+    def __init__(self, folder: str, image_extension: str = "jpg", use_gt_intrinsics: bool = True) -> None:
+        """Initializes to load from a specified folder on disk
+
+        Args:
+            folder: the base folder for a given scene
+            image_extension: (optional)extension for the image files. Defaults to 'jpg'.
+            use_gt_intrinsics: whether to use ground truth intrinsics
+        """
+        self.use_gt_intrinsics = use_gt_intrinsics
+
+        # fetch all the file names in /images folder
+        search_path = os.path.join(folder, "images", f"*.{image_extension}")
+
+        self.image_paths = glob.glob(search_path)
+
+        # sort the file names
+        self.image_paths.sort()
+        self.num_imgs = len(self.image_paths)
+
+        file_path = os.path.join(folder, "data.mat")
```

Comment:
What file is this? Looks like intrinsics from the rest of the code, but would be useful to be specific in the variable name.

Reply:
contains 3x4 camera projection matrices -- updated as `cam_matrices_fpath`